### ISSUE TITLE
False positive for constant naming style inside __main__ block

### ISSUE THREAD

**Issue by o-alquimista** [pylint-dev/pylint#10766]
### Bug description

In file `__main__.py`, I have this:

```python
def main() -> int:
    # ...

if __name__ == '__main__':
    exit_code = main()
    raise SystemExit(exit_code)
```

### Configuration

```ini
Not using a configuration file. All default options.
```

### Command used

```shell
pylint basketcase.__main__
```

### Pylint output

```python
src/basketcase/__main__.py:385:4: C0103: Constant name "exit_code" doesn't conform to UPPER_CASE naming style (invalid-name)
```

### Expected behavior

Pylint thinks `exit_code` should be named `EXIT_CODE`, as if it were a constant. The function `main` returns an integer that can currently be 1 (error) or 0 (no error).

### Pylint version

```shell
pylint 4.0.4
astroid 4.0.2
Python 3.12.3 (main, Nov  6 2025, 13:44:16) [GCC 13.3.0]
```

### OS / Environment

elementary OS 8 (based on Ubuntu 24.04).
VSCodium terminal, version 1.106.37943.

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. I agree with this one, it's been annoying me for 15 years, but we're only recently have been making invalid-name usable, so maybe that's why it resurfaces now. I don't think the module level variable defined in ``if __name__ == '__main__':`` are intended to be constants. Using them outside of the conditional itself would be prone to a lot of very strange and unexpected behavior (program doesn't behave the same way when launched from a particular file => bad). What do you think @jacobtylerwalls ?

**Comment by jacobtylerwalls** [comment]
Happy to accept this to specialize the `__name__` block. I think this was part of the ask in #10700.
